src/common/zeros_like.jl

```diff
@@ -0,0 +1,7 @@
+# Create an array of same type as X filled with zeros, minimizing the number
+# of allocations.
+function zeros_like(X::AT, n, m) where AT <: AbstractArray
+    Z = similar(X, n, m)
+    Z .= 0
+    Z
+end
```

Comment:
I would make this more general directly:
```suggestion
function zeros_like(X::AbstractArray, T::Type=eltype(X), dims::Integer=size(array)...)
    Z = similar(X, T, dims...)
    Z .= 0
    Z
end
zeros_like(X::AbstractArray, dims::Integer...) = zeros_like(X, eltype(X), dims...)
zeros_like(X::Array, T::Type=eltype(X), dims::Integer=size(array)...) = zeros(T, dims...)
```

Reply:
This doesn't seem to work because of  the default value for `dims`.
I'm a bit confused by the slurping operator. I understand it's purpose, but I don't find the syntax very clear. From what I understand by reading the docs, writing `dims::Integer...` means we want n integers to be merged into a tuple called dims: so implicitly, `dims` is here a Tuple of Integers. Is this correct? If it is, then it would explain why we can't write `dims::Integer=size(X)` as `size(X)` is a Tuple, not an Integer.
What syntax can use to explicitly say that we want `dims` (the tuple) to be `size(X)` (a tuple) by default? Or is there a way to flatten `size(X)`?

Reply:
> This doesn't seem to work because of the default value for dims.

Indeed, I had some typos in my code, corrected now.

> so implicitly, dims is here a Tuple of Integers. Is this correct?

yes

> Or is there a way to flatten size(X)?

Yes, more `...` ;), see the updated code snippet.

Reply:
Oh, ok, makes sense!